module/documents/actor/actor.mjs

```diff
@@ -1021,17 +1021,15 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
 
     const rollData = this.getRollData({deterministic: true});
 
-    let amount = damages.reduce((total, d) => {
+    damages.forEach(d => {
       // Skip damage types with immunity
-      if ( !ignore("immunity", d.type) && hasEffect("di", d.type, d.properties) ) return total;
-
-      let value = d.value;
+      if ( !ignore("immunity", d.type) && hasEffect("di", d.type, d.properties) ) return;
```

Comment:
This will end up applying the full damage for any immunities.

```suggestion
      if ( !ignore("immunity", d.type) && hasEffect("di", d.type, d.properties) ) {
        d.value = 0;
        return;
      }
```